### PR TITLE
esp32: Fix GPIO Pull-Up/Pull-Down using RTC GPIO

### PR DESCRIPTION
**Pull request by acassis**
## Summary
esp32: Fix GPIO Pull-Up/Pull-Down using RTC GPIO
## Impact
Users will be able to use pull-up and pull-down on pins weren't supporting it
## Testing
ESP32-Devkitc

### CODE REVIEW

File: arch/xtensa/src/esp32/esp32_gpio.c [hunk 1]
```diff
@@ -178,7 +193,71 @@ int esp32_configgpio(int pin, gpio_pinattr_t attr)
...
+                {
+                  /* First, disable PU/PD */
+
+                  regval &= ~SPECIAL_RTC_PU_BIT;
+                  regval &= ~SPECIAL_RTC_PD_BIT;
+
+                  /* Enable PU/PD, if needed */
+
+                  regval |= (en_pu == true) ? SPECIAL_RTC_PU_BIT : 0;
+                  regval |= (en_pd == true) ? SPECIAL_RTC_PD_BIT : 0;
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/esp32_gpio.c:242]:
```suggestion
                  regval |= en_pu ? SPECIAL_RTC_PU_BIT : 0;
                  regval |= en_pd ? SPECIAL_RTC_PD_BIT : 0;
```
Both `en_pu` and `en_pd` are booleans, so the comparison with another boolean is not necessary.

File: arch/xtensa/src/esp32/esp32_gpio.c [hunk 2]
```diff
@@ -178,7 +193,71 @@ int esp32_configgpio(int pin, gpio_pinattr_t attr)
...
+                {
+                  /* First, disable PU/PD */
+
+                  regval &= ~DEFAULT_RTC_PU_BIT;
+                  regval &= ~DEFAULT_RTC_PD_BIT;
+
+                  /* Enable PU/PD, if needed */
+
+                  regval |= (en_pu == true) ? DEFAULT_RTC_PU_BIT : 0;
+                  regval |= (en_pd == true) ? DEFAULT_RTC_PD_BIT : 0;
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/esp32_gpio.c:254]:
```suggestion
                  regval |= en_pu ? DEFAULT_RTC_PU_BIT : 0;
                  regval |= en_pd ? DEFAULT_RTC_PD_BIT : 0;
```
Both `en_pu` and `en_pd` are booleans, so the comparison with another boolean is not necessary.

File: arch/xtensa/src/esp32/esp32_gpio.c (hunk 1 above)
Comment (acassis) [reply arch/xtensa/src/esp32/esp32_gpio.c:242]: True! hehehe

**Comment by acassis** [comment]
@Ouss4 / @xiaoxiang781216 could you please merge?

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**